docs/proposals/2019-02-08 - Recommended View.md

```diff
@@ -0,0 +1,69 @@
+
+NOTE: Example values are just a suggestion and open to other names. The idea I want to convey here is that they are not naming specific applications, but rather specific methods of rendering.
+
+# Example 1
+
+Graphics team has created a static article with lots of bespoke animations and user interactive elements. They've also created an ANS version that is used for publishing to Apple News, but does not contain the same interactive elements. The website team has redirected the url to the static content. The Graphics team would like the native apps to open a webview to maximize the user experience, but also publish automatically to Apple News.
+
+```
+{
+  "type": "story",
+  "version": "0.5.9",
+  //...
+  "recommended_view": ["www"]
+}
+```
+
+Native application reading this can use the information to render summaries, alerts, etc, but when opening the article, the app will pop-up a webview. Alternatively, if this has been downloaded to the app but the user has no network they can render the native version. Apple News publisher can translate this into the Apple News format as a hard link to a web page doesn't make sense.
+
+# Example 2
+
+The team developing sponsored content has made a AR experience that is only supported by Native Apps. The website rendering layer sees this field and displays the content, still getting eyes on the sponsored content. The website team has developed a feature that can text an article to a user's phone so when they see this field/value, the site rendering will put a banner at the top of the page that says "This experience is best on your mobile device! Do you want to send it to your phone?"
+
+```
+{
+  "type": "story",
+  "version": "0.5.9",
+  //...
+  "recommended_view": ["native-app"]
+}
+```
+
+# Concerns
+
+## Does this put too much rendering concern in the document?
+
+Yes, but so does using `raw_html` or providing different rendering content than contained with the document. This information will at least make it explicit that document is incomplete as opposed to situations were the missing or `raw_html` content is merely additive.
+
+# Alternatives Considered
+
+## Why not simply not render any content that has the problematic `raw_html`?
+
+We already make decisions about whether an article is able to be viewed in our applications, but given the problem it is not always easy to tell. Often we or the content creators want a way to override the default behavior.
+
+## Why not just have a flag to force a webview?
+
+That would work for our immediate needs too, but when thinking about that option, I felt like it was serving a highly specific need. The examples above are intended to show how this concept might apply to other cases, and this was an effort to make a more useful solution.
```

Comment:
If the only two views are `www` and `native-app`, I might prefer solving the specific problem (ex: `"relies_on_raw_html": true` or `raw_html_is_central` or whatever).

Either way, I agree that this was inevitable once `raw_html` was included.

Reply:
As stated in the part you quoted, open to that too. I think this proposal offers a little more flexibility. The AR example is meant to convey a scenario that the flag doesn't cover. But if collectively that feels like over-engineering the problem I am happy to drop it. I am not a fan of your names, though, and I think they hide the true intent which is to force a webview if available. The name `webview_prefered` is more honest to the intent, though I don't like it either and think is at odds with the rendering agnostic philosophy. That is what started me on this more flexible version which might be more palatable for the purity of spec.